### ISSUE TITLE
[BUG] Unexpected Type Inference in PathExpand

### ISSUE THREAD

**Issue by BingqingLyu** [alibaba/GraphScope#3730]
**Describe the bug**

On LDBC Dataset (sf=0.1), we can run the following two queries:
```
# gremlin tests
gremlin> g.V().hasLabel("PERSON").has("id",933).out("5..6").endV().hasLabel("PLACE").has("id",999).count()
==>392
```
while a simiilar query in cypher is:
```
# cypher tests
@neo4j> Match (a:PERSON{id:933})-[c*5..6]->(b:PLACE{id:999})
        return a.id, b.id
        Limit 10;
+----------+
| id | id0 |
+----------+
+----------+
```
The results in the Cypher is empty, which is not as expected. The reason is that, in this case, the type inference tells that the type of `c` should be `isLocatedIn` (from the edge type that `Person-isLocatedIn-Place`). However, in PathExpand, this is not as expected. The path could be like `Person-Knows[4..5]-Person-IsLocatedIn-Place`, etc.

**Comment by BingqingLyu** [comment]
Fixed in #3837

**Comment by BingqingLyu** [comment]
There leaves a bug in the type infer of GetV in PathExpandBase.

For example, running the following query on LDBC dataset,
```
MATCH(a)-[*1..2]->(b) RETURN COUNT(a) AS pathCnt;
```

The inferred type of PathExpandBase within PathExpand is:
`ExpandEdge` type ids: {0,1,3,5,6,7,10,11,12,13,15,16,17,21,22};
`GetV` type ids: {0,1,2,3,5,6,7}

While here the `GetV` types missed 4, i.e., `Forum`, while `Forum` is also a valid type for the expanded vertices.
